controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -141,6 +141,20 @@ func (r *BareMetalHostReconciler) Reconcile(ctx context.Context, request ctrl.Re
 		}
 	}
 
+	// If DataImage exists, add its ownerReference
+	dataImage := &metal3api.DataImage{}
+	err = r.Get(ctx, request.NamespacedName, dataImage)
+	if !(err != nil || ownerReferenceExists(host, dataImage)) {
+		if err := controllerutil.SetControllerReference(host, dataImage, r.Scheme()); err != nil {
+			return ctrl.Result{}, fmt.Errorf("could not set bmh as controller, %w", err)
+		}
+		if err := r.Update(ctx, dataImage); err != nil {
+			return ctrl.Result{}, fmt.Errorf("failure updating dataImage status, %w", err)
```

Comment:
This seems pretty dangerous, as any failure at all to do with the DataImage (e.g. missing RBAC permissions) will now prevent the BMH itself from ever getting reconciled.

Reply:
Makes sense. Would ignoring and continuing make sense here, since the failure might potentially block bmh reconciling from happening ?